<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>官文亮</author>
 <class>SerialToolClass</class>
 <widget class="QMainWindow" name="SerialToolClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>426</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>SerialTool</string>
  </property>
  <property name="windowIcon">
   <iconset resource="serialtool.qrc">
    <normaloff>:/SerialTool/images/icon.ico</normaloff>:/SerialTool/images/icon.ico</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="maximumSize">
    <size>
     <width>16777215</width>
     <height>16777215</height>
    </size>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <property name="spacing">
     <number>3</number>
    </property>
    <property name="leftMargin">
     <number>1</number>
    </property>
    <property name="topMargin">
     <number>1</number>
    </property>
    <property name="rightMargin">
     <number>1</number>
    </property>
    <property name="bottomMargin">
     <number>1</number>
    </property>
    <item>
     <widget class="QWidget" name="portConfigWidget" native="true">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <property name="leftMargin">
        <number>2</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>2</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QLabel" name="label">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>端口：</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBoxPortNum">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>120</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>120</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">width: 120px;</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_2">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>波特率：</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBoxBaudRate">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="currentText">
          <string>1200</string>
         </property>
         <item>
          <property name="text">
           <string>1200</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>2400</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>4800</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>9600</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>19200</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>38400</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>57600</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>115200</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>串口调试助手</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <property name="spacing">
         <number>2</number>
        </property>
        <property name="leftMargin">
         <number>2</number>
        </property>
        <property name="topMargin">
         <number>2</number>
        </property>
        <property name="rightMargin">
         <number>2</number>
        </property>
        <property name="bottomMargin">
         <number>2</number>
        </property>
        <item>
         <widget class="QTextEdit" name="textEditRead">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <widget class="QWidget" name="widget" native="true">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_4">
             <property name="spacing">
              <number>3</number>
             </property>
             <property name="leftMargin">
              <number>4</number>
             </property>
             <property name="topMargin">
              <number>4</number>
             </property>
             <property name="rightMargin">
              <number>2</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QGroupBox" name="groupBox">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="title">
                <string>接收设置</string>
               </property>
               <layout class="QGridLayout" name="gridLayout_2">
                <property name="bottomMargin">
                 <number>4</number>
                </property>
                <property name="verticalSpacing">
                 <number>5</number>
                </property>
                <item row="1" column="0">
                 <widget class="QCheckBox" name="checkBox_2">
                  <property name="enabled">
                   <bool>false</bool>
                  </property>
                  <property name="text">
                   <string>自动换行</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="1">
                 <widget class="QRadioButton" name="portReadHex">
                  <property name="text">
                   <string>Hex</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="0">
                 <widget class="QRadioButton" name="portReadAscii">
                  <property name="text">
                   <string>ASCII</string>
                  </property>
                  <property name="checked">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
             <item>
              <widget class="QGroupBox" name="groupBox_2">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="title">
                <string>发送设置</string>
               </property>
               <layout class="QGridLayout" name="gridLayout">
                <property name="rightMargin">
                 <number>4</number>
                </property>
                <property name="bottomMargin">
                 <number>4</number>
                </property>
                <property name="verticalSpacing">
                 <number>5</number>
                </property>
                <item row="1" column="1">
                 <widget class="QRadioButton" name="portWriteHex">
                  <property name="text">
                   <string>Hex</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="0">
                 <widget class="QCheckBox" name="resendBox">
                  <property name="text">
                   <string>重复发送</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="3">
                 <widget class="QLabel" name="label_3">
                  <property name="text">
                   <string>ms</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="1">
                 <widget class="QSpinBox" name="spinBoxStepTime">
                  <property name="minimum">
                   <number>10</number>
                  </property>
                  <property name="maximum">
                   <number>120000</number>
                  </property>
                  <property name="value">
                   <number>1000</number>
                  </property>
                 </widget>
                </item>
                <item row="1" column="0">
                 <widget class="QRadioButton" name="portWriteAscii">
                  <property name="text">
                   <string>ASCII</string>
                  </property>
                  <property name="checked">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <layout class="QGridLayout" name="gridLayout_3">
            <item row="1" column="1">
             <widget class="QPushButton" name="sendButton">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="toolTip">
               <string>Ctrl + Enter</string>
              </property>
              <property name="text">
               <string>发送</string>
              </property>
              <property name="shortcut">
               <string>Ctrl+Return</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0" colspan="2">
             <widget class="QTextEdit" name="textEditWrite">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Ignored">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QComboBox" name="comboBox"/>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>串口示波器</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <property name="spacing">
         <number>2</number>
        </property>
        <property name="leftMargin">
         <number>2</number>
        </property>
        <property name="topMargin">
         <number>2</number>
        </property>
        <property name="rightMargin">
         <number>2</number>
        </property>
        <property name="bottomMargin">
         <number>2</number>
        </property>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <widget class="QCustomPlot" name="customPlot" native="true">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>200</width>
              <height>100</height>
             </size>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QScrollBar" name="horizontalScrollBar">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="pageStep">
             <number>100</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_9">
          <item>
           <widget class="QListWidget" name="channelList">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Ignored" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Y轴上限：</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="yRateUpperBox">
            <property name="minimum">
             <double>-100000.000000000000000</double>
            </property>
            <property name="maximum">
             <double>100000.000000000000000</double>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Y轴下限：</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="yRateLowerBox">
            <property name="minimum">
             <double>-100000.000000000000000</double>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>X轴采样数：</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="xRangeBox">
            <property name="currentText">
             <string>10</string>
            </property>
            <property name="maxVisibleItems">
             <number>20</number>
            </property>
            <item>
             <property name="text">
              <string>10</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>20</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>25</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>50</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>100</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>200</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>250</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>500</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>1000</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>2000</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>2500</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>5000</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>10000</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="holdRxOscBox">
            <property name="toolTip">
             <string>串口示波器在其他选项卡下也会接收数据</string>
            </property>
            <property name="text">
             <string>保持接收</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>600</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>文件</string>
    </property>
    <addaction name="actionSave"/>
    <addaction name="separator"/>
    <addaction name="actionClose"/>
   </widget>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>设置</string>
    </property>
    <addaction name="actionOption"/>
   </widget>
   <widget class="QMenu" name="menu_3">
    <property name="title">
     <string>编辑</string>
    </property>
    <addaction name="openPortAction"/>
    <addaction name="portRunAction"/>
    <addaction name="clearAction"/>
    <addaction name="separator"/>
    <addaction name="portSetAction"/>
   </widget>
   <widget class="QMenu" name="menu_4">
    <property name="title">
     <string>视图</string>
    </property>
    <addaction name="actionVisibleTab0"/>
    <addaction name="actionVisibleTab1"/>
    <addaction name="separator"/>
    <addaction name="actionVisibleToolbar"/>
    <addaction name="actionVisibleStatusBar"/>
   </widget>
   <widget class="QMenu" name="menu_5">
    <property name="title">
     <string>帮助</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="menu_3"/>
   <addaction name="menu_4"/>
   <addaction name="menu_2"/>
   <addaction name="menu_5"/>
  </widget>
  <widget class="QToolBar" name="toolBar1">
   <property name="windowTitle">
    <string>工具栏</string>
   </property>
   <property name="allowedAreas">
    <set>Qt::BottomToolBarArea|Qt::TopToolBarArea</set>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="portSetAction"/>
   <addaction name="separator"/>
   <addaction name="openPortAction"/>
   <addaction name="portRunAction"/>
   <addaction name="clearAction"/>
   <addaction name="separator"/>
   <addaction name="actionOption"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="serialtool.qrc">
     <normaloff>:/SerialTool/images/save.ico</normaloff>:/SerialTool/images/save.ico</iconset>
   </property>
   <property name="text">
    <string>保存</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionOption">
   <property name="icon">
    <iconset resource="serialtool.qrc">
     <normaloff>:/SerialTool/images/config.ico</normaloff>:/SerialTool/images/config.ico</iconset>
   </property>
   <property name="text">
    <string>选项</string>
   </property>
  </action>
  <action name="openPortAction">
   <property name="icon">
    <iconset resource="serialtool.qrc">
     <normaloff>:/SerialTool/images/connect.png</normaloff>:/SerialTool/images/connect.png</iconset>
   </property>
   <property name="text">
    <string>打开端口</string>
   </property>
   <property name="shortcut">
    <string>Alt+S</string>
   </property>
  </action>
  <action name="portRunAction">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="serialtool.qrc">
     <normaloff>:/SerialTool/images/pause.ico</normaloff>
     <activeoff>:/SerialTool/images/pause.png</activeoff>:/SerialTool/images/pause.ico</iconset>
   </property>
   <property name="text">
    <string>暂停发送/接收</string>
   </property>
   <property name="shortcut">
    <string>Alt+R</string>
   </property>
  </action>
  <action name="clearAction">
   <property name="icon">
    <iconset resource="serialtool.qrc">
     <normaloff>:/SerialTool/images/clear.png</normaloff>:/SerialTool/images/clear.png</iconset>
   </property>
   <property name="text">
    <string>清空数据</string>
   </property>
   <property name="shortcut">
    <string>Alt+C</string>
   </property>
  </action>
  <action name="portSetAction">
   <property name="icon">
    <iconset resource="serialtool.qrc">
     <normaloff>:/SerialTool/images/port config.png</normaloff>:/SerialTool/images/port config.png</iconset>
   </property>
   <property name="text">
    <string>端口参数设置</string>
   </property>
   <property name="toolTip">
    <string>端口设置</string>
   </property>
  </action>
  <action name="actionClose">
   <property name="icon">
    <iconset resource="serialtool.qrc">
     <normaloff>:/SerialTool/images/exit.ico</normaloff>:/SerialTool/images/exit.ico</iconset>
   </property>
   <property name="text">
    <string>退出</string>
   </property>
  </action>
  <action name="actionVisibleToolbar">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>工具栏</string>
   </property>
  </action>
  <action name="actionVisibleTab0">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>串口调试助手</string>
   </property>
   <property name="toolTip">
    <string>显示串口调试助手</string>
   </property>
  </action>
  <action name="actionVisibleTab1">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>串口示波器</string>
   </property>
   <property name="toolTip">
    <string>串口示波器</string>
   </property>
  </action>
  <action name="actionVisibleStatusBar">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>状态栏</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>关于</string>
   </property>
   <property name="shortcut">
    <string>F1</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="serialtool.qrc"/>
 </resources>
 <connections/>
</ui>
